src/main/java/uk/gov/hmcts/reform/sscs/service/DwpAddressLookupService.java

```diff
@@ -85,21 +86,27 @@ private Address buildAddress(OfficeAddress dwpAddress) {
     }
 
     public String getDwpRegionalCenterByBenefitTypeAndOffice(String benefitType, String dwpIssuingOffice) {
-
         Optional<OfficeMapping> officeMapping = getDwpMappingByOffice(benefitType, dwpIssuingOffice);
-
-        return PIP.name().equalsIgnoreCase(benefitType)
-                ? officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()).orElse(null) :
-                officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+        return getDwpRegionalCenterByBenefitType(benefitType, officeMapping);
     }
 
     public String getDefaultDwpRegionalCenterByBenefitTypeAndOffice(String benefitType) {
-
         Optional<OfficeMapping> officeMapping = getDefaultDwpMappingByBenefitType(benefitType);
+        return getDwpRegionalCenterByBenefitType(benefitType, officeMapping);
+    }
 
-        return PIP.name().equalsIgnoreCase(benefitType)
-                ? officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()).orElse(null) :
-                officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+    private String getDwpRegionalCenterByBenefitType(String benefitType, Optional<OfficeMapping> officeMapping) {
+        Benefit benefit;
+        try {
+            benefit = Benefit.getBenefitByCode(benefitType);
+            if (benefit.isHasDwpRegionCentre()) {
+                return officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()).orElse(null);
+            } else {
+                return officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+            }
+        } catch (BenefitMappingException e) {
```

Comment:
An informational comment, based on a previous comment on this PR. There's no need to change this.
If we did have a getBenefitByCodeOpt method such as below
```java
    public static Optional<Benefit> getBenefitByCodeOpt(String code) {
        Optional<Benefit> benefitByShortName = findBenefitByShortName(code);
        if (benefitByShortName.isEmpty()) {
            return findBenefitByDescription(code);
        }
        return benefitByShortName;
    }
```

Then a functional way to write this method would be like
```java
private String getDwpRegionalCenterByBenefitType(String benefitType, Optional<OfficeMapping> officeMapping) {
        return Benefit.getBenefitByCodeOpt(benefitType)
                .filter(Benefit::isHasDwpRegionCentre)
                .flatMap(benefit -> officeMapping.map(mapping -> mapping.getMapping().getDwpRegionCentre()))
                .orElse(officeMapping.map(mapping -> mapping.getMapping().getCcd())
                        .orElse(null));
    }
```
It does take a while for non-functional devs to read this, since the method as is it is now is more familiar.